lib/reader/src/parser.rs

```diff
@@ -1653,4 +1649,39 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn instruction_encoding() {
+        use cretonne::ir::Inst;
+        use cretonne::entity_map::EntityRef;
+
+        let test_file = parse_test("isa riscv supports_m
```

Comment:
```rust
       let test_file = parse_test("isa riscv supports_m
```

> So would all I need to do be drop that IR into a file `filetests/parser/instruction_encoding.rs`?

Yes, and then add `check:` lines to verify that when the function is printed out again, all the information is there.

See the existing tests in that directory and http://cretonne.readthedocs.io/en/latest/testing.html#test-cat

> Is squashing commits into one a GitHub feature? Still pretty new to pull requests

It’s a git feature. Use `git rebase -i` to squash all of your commits into one locally, then push to your branch, and the pull request will be updated.

You may have to `git push -f` because you’re rewriting history instead of just adding new commits.